vortex-file/src/read/layouts/columnar.rs

```diff
@@ -303,6 +305,41 @@ impl LayoutReader for ColumnarLayoutReader {
             Ok(Some(BatchRead::ReadMore(messages)))
         }
     }
+
+    fn read_metadata(&self) -> VortexResult<MetadataRead> {
+        let mut in_progress_metadata = self
+            .in_progress_metadata
+            .write()
+            .unwrap_or_else(|e| vortex_panic!("lock is poisoned: {e}"));
+        let mut messages = Vec::default();
+
+        for (name, child_reader) in self.names.iter().zip(self.children.iter()) {
+            match child_reader.read_metadata()? {
+                MetadataRead::Batches(data) => {
+                    in_progress_metadata.insert(name.clone(), data[0].clone());
+                }
+                MetadataRead::ReadMore(rm) => {
+                    messages.extend(rm);
+                }
+                MetadataRead::None => {
+                    in_progress_metadata.insert(name.clone(), None);
+                }
+            }
+        }
+
+        // We're done reading
+        if messages.is_empty() {
+            let child_arrays = self
+                .names
+                .iter()
+                .map(|name| in_progress_metadata[name].clone()) // TODO(Adam): Some columns might not have statistics
```

Comment:
right now we'll panic if there's one missing... but also, I think all of the children are guaranteed to exist if messages is empty? assuming that `MetadataRead::ReadMore` can't return empty messages to read